tree/treeplayer/inc/ROOT/TDFUtils.hxx

```diff
@@ -35,6 +35,16 @@ namespace ROOT {
 namespace Experimental {
 template <int D, typename P, template <int, typename, template <typename> class> class... S>
 class THist;
+
+namespace TDF {
+struct SnapshotOptions {
+  Int_t autosave   = 0;
```

Comment:
why setting `autosave` instead of `autoflush`? I read the docs but I don't understand the difference -- except that clustering is governed by `autoflush`, so maybe that's the parameter that users would most often use..? not sure :)

Reply:
Autosave is a multiple of autoflush. We probably want to only write to the file when the TTree would have flushed to disk, and not just to baskets. The change is to reflect this.

Reply:
what question does `autosave` answer?

e.g. I understand `autoflush` as the answer to the question "how frequently do you want to create a cluster?" (answer: every 100 entries, or every 30 megabytes...)

Reply:
> why setting autosave instead of autoflush?

Good question :) ... it is likely @bluehood  is right.  Either way, this commit changes from AutoFlush to AutoSave but does not provide any explanation/reasoning.

Reply:
> Autosave is a multiple of autoflush. We probably want to only write to the file when the TTree would have flushed to disk, and not just to baskets. The change is to reflect this.

Yes, but the customization of the cluster size is 'more' important as it directly affect memory use and the 'unit size' when reading back.